src/components/webPlayback.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+const track = {
+  name: "",
+  album: {
+    images: [
+      { url: "" }
+    ]
+  },
+  artists: [
+    { name: "" }
+  ]
+};
+
+
+class WebPlayback extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      is_paused: false,
+      is_active: false,
+      player: undefined,
+      is_playing_left_track: true,
+      left_side_track: track,
+      right_side_track: track,
+      previous_track: false
+    }
+
+    this.handlePrevSong = this.handlePrevSong.bind(this);
+    this.handleNextSong = this.handleNextSong.bind(this);
+  }
+
+  componentDidMount() {
+    const script = document.createElement("script");
+    script.src = 'https://sdk.scdn.co/spotify-player.js';
+    script.async = true;
+    document.body.appendChild(script);
+
+    window.onSpotifyWebPlaybackSDKReady = () => {
+      const player = new window.Spotify.Player({
+        name: 'Web Playback SDK',
+        getOAuthToken: cb => cb(this.props.token),
+        volume: 0.5
+      });
+
+      this.setState({ player: player });
+      
+      this.state.player.addListener('ready', ({ device_id }) => {
+        console.log('Ready with Device ID', device_id);
+      });
+
+      this.state.player.addListener('not_ready', ({ device_id }) => {
+        console.log('Device ID has gone offline', device_id);
+      });
+
+      this.state.player.addListener('player_state_changed', ( state => {
+        if (!state) {
+          return;
+        }
+
+        if (this.state.is_playing_left_track) {
+          this.setState({ left_side_track: state.track_window.current_track, right_side_track: state.track_window.next_tracks[0], is_paused: state.paused });
+        } else {
+          this.setState({ left_side_track: state.track_window.previous_tracks[1], right_side_track: state.track_window.current_track, is_paused: state.paused });
+        }
+
+        this.state.player.getCurrentState().then( state => {
+          (!state) ? this.setState({ is_active: false }) : this.setState({ is_active: true });
+        });
+      }));
+
+      this.state.player.connect();
+    };
+  };
+  
+  handlePrevSong() {
+    if (!this.state.is_playing_left_track) {
+      this.state.player.previousTrack();
+      this.setState({ is_playing_left_track: true });
+    } else {
+      this.state.player.previousTrack();
+      this.setState({ is_playing_left_track: false });
+    }
```

Comment:
because `this.state.player.previousTrack();` gets called in both conditions, you should just extract it out

```javascript
this.state.player.previousTrack();
if() { ... }
else { ... }
```

Reply:
furthermore, you can simplify this since you're really just setting `is_playing_left_track` on state to the opposed value of `this.state. is_playing_left_track`:

```javascript
handlePrevSong() {
  this.state.player.previousTrack();
  this.setState({ is_playing_left_track: !this.state.is_playing_left_track});
}
```